examples/cpp/20_native_dpcpp_kernel/main.cpp

```diff
@@ -0,0 +1,109 @@
+#include <iostream>
+
+#include <occa.hpp>
+
+//---[ Internal Tools ]-----------------
+// Note: These headers are not officially supported
+//       Please don't rely on it outside of the occa examples
+#include <occa/internal/utils/cli.hpp>
+#include <occa/internal/utils/testing.hpp>
+//======================================
+
+occa::json parseArgs(int argc, const char **argv);
+
+
+int main(int argc, const char **argv) {
+  occa::json args = parseArgs(argc, argv);
+
+  int entries = 16;
+
+  int *a  = new int[entries];
+  int *b  = new int[entries];
+  int *ab = new int[entries];
+
+  for (int i = 0; i < entries; ++i) {
+    a[i]  = i;
+    b[i]  = i;
+    ab[i] = 0;
+  }
+
+  // Setup the platform and device IDs
+  occa::properties deviceProps;
+  deviceProps["mode"] = "dpcpp";
+  deviceProps["platform_id"] = (int) args["options/platform-id"];
+  deviceProps["device_id"] = (int) args["options/device-id"];
+
+  occa::device device(deviceProps);
+  // Allocate memory on the device
+  occa::memory o_a = device.malloc<int>(entries);
+  occa::memory o_b = device.malloc<int>(entries);
+  occa::memory o_ab = device.malloc<int>(entries);
+
+  // Compile a regular DPCPP kernel at run-time
+  occa::properties kernelProps;
+  kernelProps["okl/enabled"] = false;
+ 
+  occa::kernel addVectors = device.buildKernel("addVectors.cpp",
+                                               "addVectors_it",
```

Comment:
Any reason this method is named differently from other examples?
```suggestion
                                               "addVectors",
```

Reply:
At some point there may have been a reason, but running the example with the name "addVectors" passes.